packages/linter/src/rules/NoIconFontIsUsed.ts

```diff
@@ -0,0 +1,103 @@
+import { Context } from "../index";
+import { Rule } from "../rule";
+
+const postcss = require("postcss");
+const safeParser = require("postcss-safe-parser");
+
+const ICON_FONT_IDENTIFIERS = [
+  {
+    className: "fa-",
+    fontFamilies: ["FontAwesome", "Font Awesome"],
+  },
+  {
+    className: "nf-",
+    fontFamilies: ["NerdFontsSymbols", "Nerd Font"],
+  },
+  {
+    className: "material-icons",
+    fontFamilies: ["Material Icons"],
+  },
+  {
+    className: "icofont-",
+    fontFamilies: ["IcoFont"],
+  },
+  {
+    className: "ociton-",
+    fontFamilies: ["Octicons"],
+  },
+];
+
+/**
+ * Checks if icon fonts are being used
+ */
+export class NoIconFontIsUsed extends Rule {
+  run({ $ }: Context) {
+    // grab all elements responding to classnames from ICON_FONT_IDENTIFIERS list
+    const iconFontCandidates = ICON_FONT_IDENTIFIERS.filter((identifier) => {
+      return $(`[class*=${identifier.className}]`).length > 0;
+    });
+
+    if (iconFontCandidates.length === 0) {
+      return this.pass();
+    }
+
+    const iconFontMatches = [];
+
+    const isIconFontDeclaration = (value) => {
+      for (const { fontFamilies } of iconFontCandidates) {
+        for (const fontFamily of fontFamilies) {
+          if (value.match(fontFamily)) {
+            return true;
+          }
+        }
+      }
+      return false;
+    };
+
+    const iconFontPlugin = postcss.plugin("postcss-icon-font-is-used", () => {
+      return (root) => {
+        root.nodes.forEach((rule) => {
+          if (rule.name === "font-face") {
+            for (const declaration of rule.nodes) {
+              if (declaration.prop === "font-family") {
+                // check if font-family matches candidate list
+                const match = isIconFontDeclaration(declaration.value);
+                if (match) {
+                  iconFontMatches.push(declaration);
+                }
+              }
+            }
+          }
+        });
+      };
+    });
+
+    // grab the amp-custom css
+    const stylesText = $("style[amp-custom]").html();
+
+    if (stylesText) {
+      postcss([iconFontPlugin])
+        .process(stylesText, {
+          from: undefined,
+          parser: safeParser,
+        })
+        .catch((err) => {
+          console.warn(`Failed to process CSS`, err.message);
+          return { css: stylesText };
+        });
+    }
+
+    if (iconFontMatches.length > 0) {
+      return this.fail("It seems like icon fonts are being used on this page.");
+    }
+
+    return this.warn("Suspicious icon font class names detected.");
```

Comment:
How likely is this case?

Reply:
Well so far I know of one case, concerning the Material Icons Font. The stylesheet is loaded via Google Web Fonts, so there is no custom css to scan for font-families, only classnames. So in this case we might give out a warning.

Reply:
In that case we could also check for `<link href="https://fonts.googleapis.com/icon?family=Material+Icons"
      rel="stylesheet">`